src/data/api.js

```diff
@@ -16,18 +18,19 @@ class WebexTeamsAPI {
 
   }
 
-  async getPerson(id) {
-    let person = null;
+  getPerson(id) {
+    const observable = from(this.axiosInstance.get(`/people/${id}`));
 
-     try {
-      person = await this.axiosInstance.get(`/people/${id}`);
+    return observable.pipe(map(person => {
```

Comment:
The observable should be part of the adapter, not the data source. The data source returns whatever it knows (object, string, promise), and the adapter is the one that transforms it into an Observable (see method definition base adapter) because that's what the components understand.

Reply:
we have not discussed about that. I still don't understand what the issue of having observables in data source.